fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java

```diff
@@ -1177,6 +1213,105 @@ public TStreamLoadPutResult streamLoadPut(TStreamLoadPutRequest request) {
+        public void run() {
+            Coordinator coord = QeProcessorImpl.INSTANCE.getCoordinator(loadId);
+            boolean notTimeout = coord.join(execTimeout);
+            // check stream load exec status
+            Status status = Status.OK;
+            if (!coord.isDone()) {
+                coord.cancel();
+                if (notTimeout) {
+                    String errMsg = coord.getExecStatus().getErrorMsg();
+                    status = new Status(ErrCode.COMMON_ERROR, "There exists unhealthy backend. " + errMsg);
+                } else {
+                    status = new Status(ErrCode.TIMEOUT, "");
+                }
+                QeProcessorImpl.INSTANCE.unregisterQuery(loadId);
+            }
+            if (!coord.getExecStatus().ok()) {
+                String errMsg = coord.getExecStatus().getErrorMsg();
+                LOG.warn("stream load failed: {}", errMsg);
+                status = new Status(ErrCode.COMMON_ERROR, errMsg);
+                QeProcessorImpl.INSTANCE.unregisterQuery(loadId);
+            }
+            PReportStreamLoadStatusRequest request = InternalService.PReportStreamLoadStatusRequest.newBuilder()
+                    .setLoadId(PUniqueId.newBuilder().setHi(loadId.hi).setLo(loadId.lo).build())
+                    .setStatus(Types.PStatus.newBuilder().addErrorMsgs(status.getErrMsg())
+                                    .setStatusCode(status.getErrCode().ordinal()).build())
+                    .build();
+
+            // get backend address
+            ImmutableMap<Long, Backend> backendMap = Env.getCurrentSystemInfo().getIdToBackend();
+            Backend be = backendMap.get(backendId);
+            TNetworkAddress address;
+            if (be == null || !be.isAlive()) {
+                LOG.warn("report stream load failed. no backend");
+                return;
+            }
+            address = new TNetworkAddress(be.getIp(), be.getBrpcPort());
+            try {
+                BackendServiceProxy.getInstance().reportStreamLoadStatus(address, request);
+            } catch (Throwable e) {
+                LOG.warn("report stream load failed.", e);
+            }
+        }
+    }
+
+    private void streamLoadPutWithSqlImpl(TStreamLoadPutRequest request) throws UserException {
+        LOG.info("receive stream load put request");
+        String loadSql = request.getLoadSql();
+        ConnectContext ctx = new ConnectContext(null);
+        ctx.setEnv(Env.getCurrentEnv());
+        ctx.setQueryId(request.getLoadId());
+        ctx.setCluster(SystemInfoService.DEFAULT_CLUSTER);
+        ctx.setCurrentUserIdentity(UserIdentity.ROOT);
+        ctx.setQualifiedUser(UserIdentity.ROOT.getQualifiedUser());
+        ctx.setThreadLocalInfo();
+        ctx.setBackendId(request.getBackendId());
+        StreamLoadTask streamLoadTask = StreamLoadTask.fromTStreamLoadPutRequest(request);
+        ctx.setStreamLoadInfo(streamLoadTask);
+        ctx.setLoadId(request.getLoadId());
```

Comment:
on line 1277, you alread set queryid to load id. so that load id is useless now.